include/rculist.h

```diff
@@ -0,0 +1,742 @@
+#ifndef _LINUX_RCULIST_H
+#define _LINUX_RCULIST_H
+
+#include <rte_atomic.h>
+#include "list.h"
+#define __rcu
+#define WRITE_ONCE(a, b) (a) = (b)
```

Comment:
Is the `WRITE_ONCE` safe enough for rculist? It's define in linux kernel source in other way. Refer to https://elixir.bootlin.com/linux/v4.18.20/source/include/linux/compiler.h#L273.